File: backend/app/models/mandi_rate.py
import uuid
from datetime import date, datetime, timezone

from sqlalchemy import Date, DateTime, Numeric, String
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base

TZ = DateTime(timezone=True)


class MandiRate(Base):
    __tablename__ = "mandi_rates"

    id: Mapped[uuid.UUID] = mapped_column(primary_key=True, default=uuid.uuid4)
    product_name: Mapped[str] = mapped_column(String(100))
    mandi_name: Mapped[str] = mapped_column(String(150))
    city: Mapped[str] = mapped_column(String(100))
    state: Mapped[str] = mapped_column(String(50))
    min_price: Mapped[float | None] = mapped_column(Numeric(10, 2))
    max_price: Mapped[float | None] = mapped_column(Numeric(10, 2))
    modal_price: Mapped[float | None] = mapped_column(Numeric(10, 2))
    unit: Mapped[str] = mapped_column(String(20), default="quintal")
    rate_date: Mapped[date] = mapped_column(Date)
    source: Mapped[str] = mapped_column(String(50), default="enam")
    created_at: Mapped[datetime] = mapped_column(
        TZ, default=lambda: datetime.now(timezone.utc)
    )

File: backend-spring/src/main/java/com/nearkart/entity/MandiRate.java
package com.nearkart.entity;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "mandi_rates")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Builder
public class MandiRate {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "product_name", nullable = false)
    private String productName;

    @Column(name = "mandi_name", nullable = false)
    private String mandiName;

    @Column(nullable = false)
    private String city;

    @Column(nullable = false)
    private String state;

    @Column(name = "min_price", precision = 10, scale = 2)
    private BigDecimal minPrice;

    @Column(name = "max_price", precision = 10, scale = 2)
    private BigDecimal maxPrice;

    @Column(name = "modal_price", precision = 10, scale = 2)
    private BigDecimal modalPrice;

    @Builder.Default
    private String unit = "quintal";

    @Column(name = "rate_date", nullable = false)
    private LocalDate rateDate;

    @Builder.Default
    private String source = "enam";

    @Column(name = "created_at", nullable = false, updatable = false)
    private OffsetDateTime createdAt;

    @PrePersist
    protected void onCreate() { createdAt = OffsetDateTime.now(); }
}
